Automattic/wp-calypso
# Gutenboarding: use Site data-store to create site

#### Changes proposed in this Pull Request
* use [Site data store](https://github.com/Automattic/wp-calypso/pull/39050) to create a site in Gutenboarding
* handle `onRequestClose` for signup modal

#### Testing instructions
* go to `/gutenboarding` and complete flow to create a website
* for logged-in users nothing should change
* for new users (incognito) there should be no more browser refresh after filling the signup form
* when having the signup modal opened, clicking outside should redirect to the previous route

Fixes https://github.com/Automattic/wp-calypso/issues/39096
